src/pdf/ShoppingList/ShoppingListPDF.tsx

```diff
@@ -0,0 +1,228 @@
+    },
+    inputText: {
+        paddingTop: "20pt",
+    },
+});
+
+const OneLine: React.FC<{ header: string; value: string }> = ({ header, value }) => {
+    return (
+        <Text style={[styles.keyText, { textTransform: "capitalize" }]}>
+            {header}: <Text style={styles.normalText}>{value}</Text>
+        </Text>
+    );
+};
+
+const TableHeadings: React.FC<{}> = () => {
+    return (
+        <View style={[styles.flexRow, styles.tableRow]}>
+            <View style={styles.tableItemDescription}>
+                <Text style={styles.keyText}>Item Description</Text>
+            </View>
+            <View style={styles.tableQuantity}>
+                <Text style={styles.keyText}>Quantity</Text>
+            </View>
+            <View style={styles.tableNotes}>
+                <Text style={styles.keyText}>Notes</Text>
+            </View>
+            <View style={styles.tableDone}>
+                <Text style={styles.keyText}>Done</Text>
+            </View>
+        </View>
+    );
+};
+
+const ItemToRow: React.FC<Item> = (item) => {
+    return (
+        <View style={[styles.flexRow, styles.tableRow]}>
+            <View style={styles.tableItemDescription}>
+                <Text style={styles.normalText}>{item.description}</Text>
+            </View>
+            <View style={styles.tableQuantity}>
+                <Text style={styles.normalText}>{item.quantity}</Text>
+            </View>
+            <View style={styles.tableNotes}>
+                <Text style={styles.normalText}>{item.notes}</Text>
+            </View>
+            <View style={styles.tableDone}></View>
+        </View>
+    );
+};
+
+const DisplayItemsList: React.FC<{ itemsList: Item[] }> = ({ itemsList }) => {
+    return (
+        <View>
+            {itemsList.map((item, index) => (
+                <ItemToRow {...item} key={index} />
+            ))}
+        </View>
+    );
+};
+
+const formatCamelCaseKey = (objectKey: string): string => {
+    return objectKey.replaceAll(/([a-z])([A-Z])/g, "$1 $2");
+};
+
+const DisplayAsBlock: React.FC<BlockProps> = (data: BlockProps) => {
+    return (
+        <View style={styles.infoCell}>
+            {Object.keys(data).map((propKey, index) => (
+                <OneLine key={index} header={formatCamelCaseKey(propKey)} value={data[propKey]} />
+            ))}
+        </View>
+    );
+};
+
+const FormatClientCell: React.FC<{ propKey: string; propValue: string }> = ({
+    propKey,
+    propValue,
+}) => {
+    return (
+        <View style={styles.infoCell}>
+            <OneLine header={formatCamelCaseKey(propKey)} value={propValue} />
+        </View>
+    );
+};
+
+const DisplayClientSummary: React.FC<ClientSummary> = (clientSummary) => {
+    return (
+        <>
+            <View style={[styles.flexRow, styles.infoBlock]}>
+                <FormatClientCell propKey="name" propValue={clientSummary.name} />
+                <FormatClientCell propKey="contact" propValue={clientSummary.contact} />
```

Comment:
Optional: but I am just thinking would it be a nice feature to implement like making the display "+44 7000 1234567"

Reply:
I think it's better to not do it than get it wrong in this case? Especially since there could be international numbers that have different formatting, and there are also different preferences (for example I prefer +44 7000 12 3456)